src/components/gataca-qrws/gataca-qrws.tsx

```diff
@@ -0,0 +1,395 @@
+   */
+  @Prop() socketEndpoint: string;
+
+  /**
+   * ***Mandatory***
+   * Maximum time window to display the session and keep the websocket connection. It's needed to ensure the socket is closed.
+   */
+  @Prop() sessionTimeout?: number = DEFAULT_SESSION_TIMEOUT;
+
+  /**
+   * [Optional]
+   * Function to send a message to the server upon socket creation
+   */
+  @Prop() wsOnOpen: (socket: WebSocket) => void;
+
+  /**
+   * **RECOMMENDED**
+   * Callback to invoke an a message has been received on the socket. It provides the socket itself and the message as parameters.
+   * If not used, the messages provided by the server on the Socket connection must conform to the WSReponse interface
+   * If used, an Event named **sessionMsg** must be triggered with a WSReponse as data
+   */
+  @Prop() wsOnMessage: (socket: WebSocket, msg: MessageEvent) => void;
+
+  /**
+   * _[Optional]_
+   * Set to enable autoload when the QR is displayed. By default it is true
+   */
+  @Prop() autostart: boolean = true;
+
+  /**
+   * _[Optional]_
+   * Set to refresh the session automatically upon expiration. By default it is false
+   */
+  @Prop() autorefresh: boolean = false;
+
+  /**
+   * **RECOMMENDED**
+   * Set to use v2 links. The create session must be providing both an authentication request and a session Id
+   */
+  @Prop() v2?: boolean = false;
+
+  /**
+   * _[Optional]_
+   * Modifies the qr headline title
+   */
+  @Prop() qrModalTitle?: string = "Quick Access";
+
+  /**
+   * _[Optional]_
+   * Modifies the Modal description
+   */
+  @Prop() qrModalDescription?: string =
+    "Sign up or sign in by scanning the QR Code with the Gataca Wallet";
+
+  /**
+   * _[Optional]_
+   * Boolean to show or not show the gataca brand title
+   */
+  @Prop() hideBrandTitle?: boolean = false;
+
+  /**
+   * _[Optional]_
+   * Display a link containing a dynamic link to invoke the wallet if closed
+   */
+  @Prop() dynamicLink?: boolean = true;
+
+  @State() sessionId?: string;
+  @State() authenticationRequest?: string;
+  @State() sessionData: any = undefined;
+  @State() result: RESULT_STATUS = RESULT_STATUS.NOT_STARTED;
+
+  async componentDidLoad() {
+    if (this.autostart) {
+      await this.display();
+    }
+  }
+
+  /**
+   * GatacaLoginCompleted event, triggered with session data upon login success
+   */
+  @Event({
+    eventName: "gatacaLoginCompleted",
+    composed: true,
+    cancelable: true,
+    bubbles: true,
+  })
+  gatacaLoginCompleted: EventEmitter;
+
+  /**
+   * GatacaLoginFailed event, triggered with error upon login failure
+   */
+  @Event({
+    eventName: "gatacaLoginFailed",
+    composed: true,
+    cancelable: true,
+    bubbles: true,
+  })
+  gatacaLoginFailed: EventEmitter;
+
+  @State() socket: WebSocket;
+
+  /**
+   * Force manually the display of a QR
+   */
+  @Method()
+  async display(): Promise<void> {
+    let socket = new WebSocket("ws://127.0.0.1:1323/ws");
```

Comment:
This is OK?

Reply:
The WS part is being tested and needs documentation.